app/addons/fauxton/components.react.jsx

```diff
@@ -84,9 +84,131 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
   });
 
 
+  var Tray = React.createClass({
+
+    getInitialState : function () {
+      return {
+        show: false
+      };
+    },
+
+    toggle : function (done) {
+      if (this.state.show) {
+        this.hide(done);
+      } else {
+        this.show(done);
+      }
+    },
+
+    componentDidMount : function () {
+      var that = this;
+      $('body').on('click.Tray-' + this.props.trayid + "-" + this._rootNodeID, function (e) {
+        var tgt = $(e.target);
+        if (that.state.show && tgt.closest('.tray').length === 0) {
+          that.hide();
+        }
+      });
+    },
+
+    componentWillUnmount : function () {
+      $('body').off('click.Tray-' + this.props.trayid + "-" + this._rootNodeID);
+    },
+
+    show : function (done) {
+      this.setState({show: true});
+      $(this.refs.myself.getDOMNode()).velocity('transition.slideDownIn', FauxtonAPI.constants.MISC.TRAY_TOGGLE_SPEED, function () {
+        if (done) {
+          done(true);
+        }
+      });
+    },
+
+    hide : function (done) {
+      var that = this;
+      $(this.refs.myself.getDOMNode()).velocity('reverse', FauxtonAPI.constants.MISC.TRAY_TOGGLE_SPEED, function () {
+        that.setState({show: false});
+        if (done) {
+          done(false);
+        }
+      });
+    },
+
+    render : function () {
+      var styleSpec = this.state.show ? {"display": "block", "opacity": 1} :  {"display": "none", "opacity": 0};
+      var classSpec = this.props.className || "";
+      classSpec += " tray";
+      return (
+        <div ref="myself" style={styleSpec} className={classSpec}>{this.props.children}</div>
+      );
+    }
+  });
+
+  var Pagination = React.createClass({
+
+    getInitialState: function () {
+      return {};
+    },
```

Comment:
actually: no - you'll break the logic, removing these three lines will crash the UI (or you just drop the possibility to set some things via state also and not just via props)